storage/scylla_api.h

```diff
@@ -0,0 +1,199 @@
+  "OORVZT"
+#define ADDRESS_3                                                               \
+  "ADDRESS300000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OORVZT"
+#define ADDRESS_4                                                               \
+  "ADDRESS400000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OORVZT"
+#define ADDRESS_5                                                               \
+  "ADDRESS500000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OORVZT"
+#define ADDRESS_6                                                               \
+  "ADDRESS600000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OORVZT"
+
+#define TRANSACTION_1                                                               \
+  "TRANSACTION100000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_2                                                               \
+  "TRANSACTION200000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_3                                                               \
+  "TRANSACTION300000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_4                                                               \
+  "TRANSACTION400000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_5                                                               \
+  "TRANSACTION500000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_6                                                               \
+  "TRANSACTION600000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_7                                                               \
+  "TRANSACTION700000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_8                                                               \
+  "TRANSACTION800000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_9                                                               \
+  "TRANSACTION900000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+
+#define TIMESTAMP_1 1
+#define TIMESTAMP_2 2
+#define TIMESTAMP_3 3
+
+#define MESSAGE_1 "MESSAGE1"
+#define MESSAGE_2 "MESSAGE2"
+#define MESSAGE_3 "MESSAGE3"
+#define MESSAGE_4 "MESSAGE4"
+#define MESSAGE_5 "MESSAGE5"
+#define MESSAGE_6 "MESSAGE6"
+
+typedef struct scylla_iota_transaction scylla_iota_transaction_t;
+
+status_t new_scylla_iota_transaction(scylla_iota_transaction_t** obj);
+void free_scylla_iota_transaction(scylla_iota_transaction_t** obj);
+
+status_t set_transaction_hash(scylla_iota_transaction_t* obj, cass_byte_t* hash, size_t length);
```

Comment:
I think because these following functions are used in `tests` only, it would be better to move them there

Reply:
These functions will be used by other programs which need to set the transaction data to save into ScyllaDB.

Reply:
If these the function that will be used in the future, hiw about move them to the commit that you will implement in the future, because I personally don't like to separate definition and declaration in different places which we did not do in this way in TA as well.

Reply:
ZMQ client I do not commit yet and other programs who want to use this interface should call those functions. These no other way to set member of struct scylla_iota_transaction.

Reply:
OK!

Reply:
But the doxygen friendly doc are missed for these functions

Reply:
Yeah..I should add doxygen docs.
Too many functions boring..QQ

Reply:
I know that xd